src/components/hosts/HostsTable.tsx

```diff
@@ -299,8 +299,13 @@ const HostsTable: React.FC<HostsTableProps> = ({ cluster, skipDisabled = false }
     [setSortBy, setOpenRows],
   );
 
-  const getHostInventory = (hostId: Host['id']): Inventory =>
-    hostRows.find((rowData) => rowData.host?.id === hostId)?.inventory;
+  let eventsModalTitle = 'Host Events';
+  if (showEventsModal) {
+    const hostRow = hostRows.find((rowData) => rowData.host?.id === showEventsModal.host);
+    const hostname =
+      hostRow?.host.requestedHostname || hostRow?.inventory?.hostname || showEventsModal.host;
+    eventsModalTitle += `: ${hostname}`;
+  }
```

Comment:
I think this would be much better to calculate when calling `setShowEventsModal` in `onViewHostEvents` as that is exactly the moment when we need the hostname. Ideally the object passed to `setShowEventsModal` would be
```
{ clusterId: Cluster['id'], hostId: Host['id'], hostname: string }
```

Reply:
done